lib/shopify_transporter/pipeline/magento/order/line_items.rb

```diff
@@ -19,14 +19,24 @@ def convert(input, record)
 
           def line_items(input)
             line_items = line_items_array(input)
-            line_items.map { |item| line_item(item) }
+            combine_associated_line_items(line_items.map { |item| line_item(item) })
+          end
+
+          def combine_associated_line_items(line_items)
+            line_items.group_by { |line_item| line_item['sku'] }.map do |_, associated_items|
+              case associated_items.size
+              when 1
+                associated_items.first
+              when 2
+                parent = associated_items.find { |x| x['product_type'] == 'configurable' }
+                child = associated_items.find { |x| x['product_type'] == 'simple' }
+                parent.merge(child.slice('name'))
+              end.except('product_type')
```

Comment:
The csv converter does just ignore it. However, all the unit tests are testing against the return of this code, which is the intermediate format before it gets passed to the CSV converter. I think it would be kinda going against the spirit of TDD to add `product_type` to all the unit tests to get them to pass.

IMO it's better to just remove it here than pass it along through the rest of the code flow, adding it to all the unit tests for all those intermediate stages, then let it get ignored by the CSV converter.

Reply:
Good points. Agreed!